app/cdap/components/WrangleHome/services/getWidgetData.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+
+export const getWidgetData = async (cbUpdateState) => {
+  const connectorTypes = await fetchConnectors();
+  const categorizedConnections = await getCategorizedConnections();
+
+  const connectorTypeWithConnections = [];
+  categorizedConnections?.forEach((value, key) => {
+    let mostUpdatedTimeStamp = value[0].updatedTimeMillis;
+    value.forEach((e) => {
+      if (mostUpdatedTimeStamp < e.updatedTimeMillis) {
+        mostUpdatedTimeStamp = e.updatedTimeMillis;
+      }
+    });
+    connectorTypeWithConnections.push({ name: key, time: mostUpdatedTimeStamp });
+  });
+  const sortedConections = connectorTypeWithConnections.slice(0);
+  sortedConections.sort((a, b) => b.time - a.time);
+
+  const connectorDataArray = [];
+  let connectorDataWithSvgArray: IConnectorArray[] = [];
+  const allConnectorsPluginProperties: Map<
+    string,
+    IConnectorDetailPayloadArray[]
+  > = getCategoriesToConnectorsMap(connectorTypes);
+  const connectionPayloadArray: IConnectorDetailPayloadArray[] = [];
+  allConnectorsPluginProperties?.forEach((connectorsArray) => {
+    if (connectorsArray.length) {
+      connectorsArray.map((item) => {
+        connectionPayloadArray.push(item);
+      });
+    }
+  });
+
+  const connectionDetailsData = await Promise.all(
+    connectionPayloadArray.map(async (item, index) => {
+      const selectedConnector = {
+        artifact: item.artifact,
+        category: item.category,
+        name: item.name,
+        type: item.type,
+      };
+      connectorDataArray.push(selectedConnector);
+      return new Promise((resolve, reject) => {
+        const response = fetchConnectionDetails(selectedConnector);
+        if (response) {
+          resolve(response);
+        }
+      });
+    })
+  );
+  const connectorWidgetJson = connectionDetailsData.map(
+    ({ connectorWidgetJSON }) => connectorWidgetJSON
+  );
+
+  connectorDataArray.map((connectorType) => {
+    let connectorTypeHasWidget = false;
+    /**
+     * Getting widget icons for connector types
+     */
+    connectorWidgetJson.map((item) => {
+      if (item['display-name'] && item['display-name'].includes(connectorType.name)) {
+        connectorDataWithSvgArray.push({
+          ...connectorType,
+          SVG: <WidgetSVG dataSrc={item?.icon?.arguments?.data} />,
+        });
+        connectorTypeHasWidget = true;
+      }
+    });
+    /**
+     * Retaining the connector types which are not part of widget api
+     */
+    if (!connectorTypeHasWidget) {
+      connectorDataWithSvgArray.push({
+        ...connectorType,
+        SVG: <WidgetSVG dataSrc={undefined} />,
+      });
+    }
+  });
+
+  connectorDataWithSvgArray = sortedConections.map((obj) =>
+    connectorDataWithSvgArray.find((item) => item.name === obj.name)
+  );
+
+  connectorDataWithSvgArray = [
+    ...new Map(connectorDataWithSvgArray.map((item) => [item.name, item])).values(),
+  ];
+  const staticCardModel = {
```

Comment:
declare this above the function instead of right here?

Reply:
created a new function to attach new cards data